src/controller/CHIPDeviceController.h

```diff
@@ -706,6 +706,25 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
      */
     CHIP_ERROR ICDRegistrationInfoReady();
 
+    /**
+     * @brief
+     *   This function is called by the upper layer application to indicate that the required terms and conditions
+     *   acknowledgements have been set. This function should be called after the terms and conditions bitmask and version
+     *   have been defined using the appropriate configuration macros and the application has gathered the necessary
+     *   acknowledgements from the user.
+     *
+     * The upper layer application should call this method once it has successfully presented and obtained acknowledgements
+     * for the required terms and conditions from the user. This indicates that the commissioning process can advance to the
+     * next stage.
+     *
+     * When the terms and conditions acknowledgements process is completed, this function will signal the readiness to proceed
+     * to the next step in the commissioning process.
+     *
+     * @return CHIP_ERROR   The return status. Returns CHIP_ERROR_INCORRECT_STATE if the function is called when the device
+     *                      is not in the correct state to accept terms and conditions acknowledgements.
+     */
+    CHIP_ERROR TermsAndConditionsAcknowledgementsReady();
```

Comment:
One other very important thing: this API surface would ensure that we can never do multiple commissionings in parallel, because nothing ties this call to a specific instance of the commissioning process.  This is even an issue or "serial" commissionings: one might get canceled and another started, then this call comes in for the first one....

This needs a better solution.

Reply:
I was originally under the impression that we supported 11-digit code that did not provide VID/PID then identified the VID/PID from an announcement from the device, and then could perform a DCL lookup. I learned that we only support 21-digit and QR code for custom commissioning flow (from Chris D.). So…. We should know the VID/PID prior to initiating the AutoCommissioner, and we can require the UX to provide the acknowledgement as a pre-condition, thus avoiding any “wait” stage.

Reply:
After discussing offline, I have removed the wait-state.